restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/Conference.java

```diff
@@ -577,4 +600,59 @@ private void updateConferenceStatus(ConferenceStateChanged.State state){
             dao.updateConferenceDetailRecordStatus(cdr);
         }
     }
+
+    /**
+     * startConferenceTimer - conference should expire after 4 hours/(configurable)
+     */
+    private void startConferenceTimer() {
+        final long conferenceTotalLifeInMillis = RestcommConfiguration.getInstance().getMain().getConferenceTimeout()*1000;
+        final long conferenceRemainingLife =  conferenceTotalLifeInMillis - conferenceAge();
+        context().setReceiveTimeout(Duration.create(conferenceRemainingLife, TimeUnit.MILLISECONDS));
+        if(logger.isInfoEnabled())
+            logger.info(String.format("conference timer started for: %s milliseconds", conferenceRemainingLife));
+    }
+
+    /**
+     * @return conference age in milli seconds
+     */
+    private long conferenceAge() {
+        if(sid != null){
+            final ConferenceDetailRecordsDao dao = storage.getConferenceDetailRecordsDao();
+            ConferenceDetailRecord cdr = dao.getConferenceDetailRecord(sid);
+            if(cdr == null){
+                if(logger.isDebugEnabled())
+                    logger.debug(String.format("Conference cdr is null for sid: %s", sid));
+            }else{
+                return DateTime.now().getMillis()-cdr.getDateCreated().getMillis();
```

Comment:
can we assume all JVM/OS nodes in the cluster are clock synced? are the ndoes configured to sync system clock form a central server regularly?

Reply:
@jaimecasero Network Time Protocol (NTP) is configured by default on Amazon Linux instances and for cloud service is up and running to make it synchronised to NTP server.